std/range/package.d

```diff
@@ -4264,6 +4348,33 @@ unittest
    }
 }
 
+unittest // Bugzilla 15860: foreach_reverse on lockstep
+{
+    auto arr1 = [0, 1, 2, 3];
+    auto arr2 = [4, 5, 6, 7];
+
+    size_t n = arr1.length -1;
+    foreach_reverse (index, a, b; lockstep(arr1, arr2, StoppingPolicy.requireSameLength))
+    {
+        assert(n == index);
+        assert(index == a);
+        assert(arr1[index] == a);
+        assert(arr2[index] == b);
+        n--;
+    }
+
+    auto arr3 = [4, 5];
+    n = 0;
+    foreach_reverse (a, b; lockstep(arr1, arr3))
+    {
+        if (n == 0)
+            assert(a == 3 && b == 5);
+        if (n == 1)
+            assert(a == 2 && b == 4);
+        n++;
```

Comment:
can't you just compare with n as index like in your examples? Looks a bit weird this way